app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,761 @@
+
+                    //Image rotation
+                    int rotation = deviceOrientation.getOrientation();
+                    bitmapImage = imageOrientation(bitmapImage, rotation);
+
+                    //save Bitmap inside internal storage and reset extracted text
+                    bitmapManager.saveBitmapToInternalStorage(bitmapImage);
+                    SharedPreferences sharedPref = getApplicationContext()
+                            .getSharedPreferences("extractedText", Context.MODE_PRIVATE);
+                    SharedPreferences.Editor editor = sharedPref.edit();
+                    editor.putString("lastExtractedText", "");
+                    editor.apply();
+
+                    Intent i = new Intent(CameraActivity.this, ResultActivity.class);
+                    startActivity(i);
+
+                }
+            };
+
+            mImageReader.setOnImageAvailableListener(readerListener, mBackgroundHandler);
+
+            //@author Leonardo Rossi
+
+            //Output surfaces
+            List<Surface> outputSurface = new ArrayList<>(2);
+            outputSurface.add(mImageReader.getSurface());
+            outputSurface.add(new Surface(mCameraTextureView.getSurfaceTexture()));
+
+            //Creation of a capture request to take a photo from the camera
+            final CaptureRequest.Builder captureBuilder = mCameraDevice.createCaptureRequest(CameraDevice.TEMPLATE_STILL_CAPTURE);
+            captureBuilder.addTarget(mImageReader.getSurface());
+            captureBuilder.set(CaptureRequest.CONTROL_MODE, CameraMetadata.CONTROL_MODE_AUTO);
+
+
+            final CameraCaptureSession.CaptureCallback captureListener = new CameraCaptureSession.CaptureCallback() {
+                @Override
+                public void onCaptureCompleted(@NonNull CameraCaptureSession session,
+                                               @NonNull CaptureRequest request, @NonNull TotalCaptureResult result) {
+                    super.onCaptureCompleted(session, request, result);
+                    createCameraPreview();
+                }
+            };
+
+            mCameraDevice.createCaptureSession(outputSurface, new CameraCaptureSession.StateCallback() {
+                @Override
+                public void onConfigured(@NonNull CameraCaptureSession cameraCaptureSession) {
+                    try {
+                        cameraCaptureSession.capture(captureBuilder.build(),captureListener, mBackgroundHandler);
+                    } catch (CameraAccessException e) {
+                        e.printStackTrace();
+                    }
+                }
+
+                @Override
+                public void onConfigureFailed(@NonNull CameraCaptureSession cameraCaptureSession)
+                {
+                }
+            },mBackgroundHandler);
+        } catch(CameraAccessException e) {
+            e.printStackTrace();
+        }
+    } //end takePhoto
+
```

Comment:
the code in this method is clean and clear and it has been commented well. I ask you why keep in the code the method onConfiguredFailed considering that is not used and has not been written.

Reply:
Thank you Mattia! To be sincere I don't know why the method onConfiguredFailed has not been removed here, but it's very likely that in a more recent commit it would have been so. I need to recall that this file comes from a rather outdated commit due to the fact that the method takePhoto() has been completely refactored by now.